mteb/tasks/InstructionRetrieval/multilingual/mFollowIR.py

```diff
@@ -0,0 +1,370 @@
+
+        # Load qrels_changed data
+        qrels_changed_data = datasets.load_dataset(
+            path,
+            f"qrels_changed-{loading_lang}",
+            cache_dir=cache_dir,
+            revision=revision,
+            trust_remote_code=True,
+        )
+        for row in qrels_changed_data[EVAL_SPLIT]:
+            if row["query-id"] not in changed_relevant_docs[lang][EVAL_SPLIT]:
+                changed_relevant_docs[lang][EVAL_SPLIT][row["query-id"]] = {
+                    row["corpus-id"]: int(row["score"])
+                }
+            else:
+                changed_relevant_docs[lang][EVAL_SPLIT][row["query-id"]][
+                    row["corpus-id"]
+                ] = int(row["score"])
+
+        # Load top_ranked data
+        top_ranked_data = datasets.load_dataset(
+            path,
+            f"top_ranked-{loading_lang}",
+            cache_dir=cache_dir,
+            revision=revision,
+            trust_remote_code=True,
+        )
+        for row in top_ranked_data["top_ranked"]:
+            if row["qid"] not in top_ranked[lang][EVAL_SPLIT]:
+                top_ranked[lang][EVAL_SPLIT][row["qid"]] = [row["pid"]]
+            else:
+                top_ranked[lang][EVAL_SPLIT][row["qid"]].append(row["pid"])
+
+    # make og_instructions and changed_instructions from queries and then turn queries into just queries
+    og_instructions = {lang: {EVAL_SPLIT: defaultdict(dict)} for lang in queries}
+    changed_instructions = {lang: {EVAL_SPLIT: defaultdict(dict)} for lang in queries}
+    queries_only = {lang: {EVAL_SPLIT: {}} for lang in queries}
+    for lang in queries:
+        for split in queries[lang]:
+            for qid in queries[lang][split]:
+                text = queries[lang][split][qid]["text"]
+                og_instructions[lang][split][text] = queries[lang][split][qid][
+                    "instruction_og"
+                ]
+                changed_instructions[lang][split][text] = queries[lang][split][qid][
+                    "instruction_changed"
+                ]
+                queries_only[lang][split][qid] = text
+
+    queries = queries_only
+
+    return (
+        corpus,
+        queries,
+        og_instructions,
+        changed_instructions,
+        og_relevant_docs,
+        changed_relevant_docs,
+        top_ranked,
+    )
+
+
+class mFollowIRCrossLingual(MultilingualTask, AbsTaskInstructionRetrieval):
+    metadata = TaskMetadata(
+        name="mFollowIRCrossLingualInstructionRetrieval",
+        description="This tasks measures retrieval instruction following ability on NeuCLIR narratives for the mFollowIR benchmark on the Farsi, Russian, and Chinese languages with English queries/instructions.",
+        reference="https://neuclir.github.io/",
+        dataset={
+            "path": "jhu-clsp/mFollowIR-cross-lingual",
+            "revision": "f85872d09da933018d64e4a7260584e536b7cf14",
+            "trust_remote_code": True,
```

Comment:
It would be great if we could avoid this. I believe @gowitheflow-1998 made a solution for it [here](https://github.com/embeddings-benchmark/mteb/issues/1090)

Reply:
Your splits look good. I think you can download them and reupload with huggingface datasets and they will be in parquet with all metadata. I did this while adding coir dataset.

Reply:
Hi @orionw sorry for the late reply! Hopefully @Samoed's solution works! If it doesn't automatically work, a backup solution will be turning the files into parquet locally first and push again, then manually adjust the readme in the HF repo to make sure each subset is reading the correct parquet file.